### PR TITLE
Add new package: icoutils

### PR DESCRIPTION
**Pull request by darmac**

### CODE REVIEW

File: var/spack/repos/builtin/packages/icoutils/package.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Icoutils(AutotoolsPackage):
+    """extract and convert Microsoft icon and cursor files."""
+
+    homepage = "http://www.nongnu.org/icoutils"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/icoutils/package.py]:
```suggestion
    homepage = "https://www.nongnu.org/icoutils"
```

File: var/spack/repos/builtin/packages/icoutils/package.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    url      = "http://nongnu.askapache.com/icoutils/icoutils-0.32.3.tar.bz2"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/icoutils/package.py]:
```suggestion
    url      = "https://nongnu.askapache.com/icoutils/icoutils-0.32.3.tar.bz2"
```

**Comment by tldahlgren** [comment]
Please use `https` for these packages.

File: var/spack/repos/builtin/packages/icoutils/package.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    version('0.32.3', sha256='17abe02d043a253b68b47e3af69c9fc755b895db68fdc8811786125df564c6e0')
+    version('0.32.2', sha256='e892affbdc19cb640b626b62608475073bbfa809dc0c9850f0713d22788711bd')
+    version('0.32.1', sha256='a9ed0fdb93e35708d9d07d0433ee1d71e07683404fa8d6e9d8aaeb6506a13fbf')
+
+    depends_on('libpng')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/icoutils/package.py]: FWIW. The dependencies can be further constrained per https://archlinux.org/packages/community/x86_64/icoutils/.

**Review by alalazo — CHANGES_REQUESTED** [review]
Can you use the `GnuMirrorPackage` mixin for this one? You may have a look at `automake` as an example.

**Comment by darmac** [comment]
Of cource, but why we should add that, I didn't found the gnu url and there's no explanation for `GNUMirrorPackage` in spack manu.

**Pull request closed without merging**